eva/experimental/ray/optimizer/rules/rules.py

```diff
@@ -55,28 +55,36 @@ def apply(self, before: LogicalExchange, context: OptimizerContext):
         yield after
 
 
-class LogicalProjectToPhysical(Rule):
+class LogicalApplyAndMergeToPhysical(Rule):
     def __init__(self):
-        pattern = Pattern(OperatorType.LOGICALPROJECT)
+        pattern = Pattern(OperatorType.LOGICAL_APPLY_AND_MERGE)
         pattern.append_child(Pattern(OperatorType.DUMMY))
-        super().__init__(RuleType.LOGICAL_PROJECT_TO_PHYSICAL, pattern)
+        super().__init__(RuleType.LOGICAL_APPLY_AND_MERGE_TO_PHYSICAL, pattern)
 
     def promise(self):
-        return Promise.LOGICAL_PROJECT_TO_PHYSICAL
+        return Promise.LOGICAL_APPLY_AND_MERGE_TO_PHYSICAL
 
     def check(self, grp_id: int, context: OptimizerContext):
         return True
 
-    def apply(self, before: LogicalProject, context: OptimizerContext):
-        after = ProjectPlan(before.target_list)
-        for child in before.children:
-            after.append_child(child)
-        upper = ExchangePlan(
-            parallelism=1,
-            ray_conf={"num_gpus": 1} if Context().gpus else {"num_cpus": 1},
+    def apply(self, before: LogicalApplyAndMerge, context: OptimizerContext):
+        apply_plan = ApplyAndMergePlan(before.func_expr, before.alias, before.do_unnest)
```

Comment:
Some functions we can not parallelize like trackers. We can add a new UDF attribute to the decorator and check here.

Reply:
I see. Just curious, why tracker is not parallelizable?

Reply:
Tracker needs tuples in sequence. For it to run in parallel, we need to implement some nontrivial stitching code.